### PR TITLE
[improve][broker] Using TopicName instead of String as the parameter for `getTopic`.

### PR DESCRIPTION
**Pull request by Technoboy-**
### Motivation

When we call broker#getTopic to get topics, the same topic requests twice, once with the domain and once without the domain, then there will be 2 instances in `topics`. Both instances are associated with the same ManagedLedger. 

Because we put the key `persistent://tn/ns/xxx` and `tn/ns/xxx` into the `topics`.  In this case, when we close the topic, another key related to the topic and the ManagedLedger may cause creating producer/consumer failure. see the below test.

### Documentation

- [x] `doc-not-needed` 
(Please explain why)

### CODE REVIEW

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java [hunk 1]
```diff
@@ -235,4 +235,27 @@ private void testSetBacklogCausedCreatingProducerFailure() throws Exception {
...
+        final String ns = "prop/ns-test";
+        admin.namespaces().createNamespace(ns, 2);
+        final String topicName = ns + "/topic-1";
+        admin.topics().createNonPartitionedTopic(String.format("persistent://%s", topicName));
+        Producer<String> producer1 = pulsarClient.newProducer(Schema.STRING).topic(topicName).create();
+        producer1.close();
+        PersistentTopic persistentTopic = (PersistentTopic) pulsar.getBrokerService().getTopic(topicName.toString(), false).get().get();
+        persistentTopic.close().join();
+        List<String> topics = new ArrayList<>();
+        topics.addAll(pulsar.getBrokerService().getTopics().keys());
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java:250]:
```suggestion
        List<String> topics = new ArrayList<>(pulsar.getBrokerService().getTopics().keys());
```

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java [hunk 2]
```diff
@@ -235,4 +235,27 @@ private void testSetBacklogCausedCreatingProducerFailure() throws Exception {
             Assert.fail("failed to create producer");
         }
     }
+
+    @Test
+    public void testGetTopic() throws Exception {
```
Comment (codelipenghui) [inline pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java:240]: One question about the test.
Why the test should be added to the `PartitionedSystemTopicTest.java`?
Looks like they are unrelated.

The `BrokerServiceTest` looks more reasonable

**Review by codelipenghui — APPROVED** [review]
Interesting problem. Does the issue always exist or introduced from a version?

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java (hunk 2 above)
Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java:240]: ok

Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java]: fixed.

File: pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java (hunk 1 above)
Comment (Technoboy-) [reply pulsar-broker/src/test/java/org/apache/pulsar/broker/systopic/PartitionedSystemTopicTest.java]: fixed

**Comment by Technoboy-** [comment]
> Interesting problem. Does the issue always exist or introduced from a version?

Seems always exists

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java [hunk 3]
```diff
@@ -1005,14 +1009,14 @@ public CompletableFuture<Optional<Topic>> getTopic(final String topic, boolean c
                 });
             }
         } catch (IllegalArgumentException e) {
-            log.warn("[{}] Illegalargument exception when loading topic", topic, e);
+            log.warn("[{}] Illegalargument exception when loading topic", topicName.toString(), e);
```
Comment (mattisonchao) [inline pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java:1012]: It looks like we don't need to call `toString` here.

**Review by mattisonchao — APPROVED** [review]
Nice catch

File: pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java (hunk 3 above)
Comment (Technoboy-) [reply pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java]: fixed.

**Review by codelipenghui — APPROVED** [review]

**Comment by nodece** [comment]
This PR has been cherry-pick to branch-2.11 by https://github.com/apache/pulsar/pull/19632

**Pull request merged**